include/psa/crypto_sizes.h

```diff
@@ -589,74 +678,394 @@
+ *
+ * \param key_type      A symmetric key type that is compatible with algorithm
+ *                      alg.
+ * \param alg           A cipher algorithm (\c PSA_ALG_XXX value such that
+ *                      #PSA_ALG_IS_CIPHER(\p alg) is true).
+ *
+ * \return              The default IV size for the specified key type and
+ *                      algorithm. If the algorithm does not use an IV,
+ *                      return 0. If the key type or cipher algorithm
+ *                      is not recognized, or the parameters are incompatible,
+ *                      return 0. An implementation can return either 0
+ *                      or a correct size for a key type and cipher algorithm
+ *                      that it recognizes, but does not support.
+ */
+#define PSA_CIPHER_IV_LENGTH(key_type, alg)                                         \
+    (PSA_ALG_IS_CIPHER(alg) &&                                                      \
+     ((key_type) & PSA_KEY_TYPE_CATEGORY_MASK) == PSA_KEY_TYPE_CATEGORY_SYMMETRIC ? \
+     (PSA_BLOCK_CIPHER_BLOCK_LENGTH(key_type) > 1 ?                                 \
+      PSA_BLOCK_CIPHER_BLOCK_LENGTH(key_type) :                                     \
+      PSA_BLOCK_CIPHER_BLOCK_MAX_SIZE) :                                            \
      0)
 
+/** The maximum size of the output of psa_cipher_encrypt(), in bytes.
+ *
+ * If the size of the output buffer is at least this large, it is guaranteed
+ * that psa_cipher_encrypt() will not fail due to an insufficient buffer size.
+ * Depending on the algorithm, the actual size of the output might be smaller.
+ *
+ * See also #PSA_CIPHER_ENCRYPT_OUTPUT_MAX_SIZE(\p input_length).
+ *
+ * \warning This function can evaluate its arguments multiple times or
+ *          zero times. Providing arguments that have side effects will
+ *          result in implementation-specific behavior, and is non-portable.
+ *
+ * \param key_type      A symmetric key type that is compatible with algorithm
+ *                      alg.
+ * \param alg           A cipher algorithm (\c PSA_ALG_XXX value such that
+ *                      #PSA_ALG_IS_CIPHER(\p alg) is true).
+ * \param input_length  Size of the input in bytes.
+ *
+ * \return              A sufficient output size for the specified key type and
+ *                      algorithm. If the key type or cipher algorithm is not
+ *                      recognized, or the parameters are incompatible,
+ *                      return 0. An implementation can return either 0 or
+ *                      a correct size for a key type and cipher algorithm
+ *                      that it recognizes, but does not support.
+ */
+#define PSA_CIPHER_ENCRYPT_OUTPUT_SIZE(key_type, alg, input_length)                     \
+    (PSA_ALG_IS_CIPHER(alg) &&                                                          \
+     ((key_type) & PSA_KEY_TYPE_CATEGORY_MASK) == PSA_KEY_TYPE_CATEGORY_SYMMETRIC ?     \
```

Comment:
We don't guarantee the value if `key_type` isn't compatible with `alg`, so it's enough to do dispatch on `alg`:
```
alg == CBC_PKCS7 ? (rounded up to block size) : IS_CIPHER(alg) ? input+iv : 0
```